---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# topR

See full documentation at [https://wuxi-nextcode.github.io/topR/](https://wuxi-nextcode.github.io/topR/)

### Installing from github using `devtools`

```{r, eval=FALSE}
devtools::install_github("wuxi-nextcode/topR")
```

## Example
In this example, we're going to demonstrate the basic usage of the topR library. 

### Load packages

First load the `gorr` package, the `tidyverse` package is recommended in general, but not required for this example

``` {r}
library(topR)
library(tidyverse)
```
``` {r, echo=FALSE}
library(ggrepel)
```

### Loading and exploring prebuilt datasets 

Load the `gwas_CD` and checking how many variants are in the datasets, since we do not want to plot with too many datapoints

```{r}
data(gwas_CD)
head(gwas_CD)
```

```{r}
paste("Number of SNPs in rhw dataset: [", length(gwas_CD$POS),"]", sep = "")
```
### Manhattan plots

Get an overview of the crohn's disease (CD) results in a Manhattan plot

```{r}
manhattan(gwas_CD)
```

get the top SNP per 10 MB

```{r}
snps_CD <- get_best_snp_per_MB(gwas_CD, thresh = 1e-09, region = 10000000)
manhattan(gwas_CD, variants=snps_CD, annotation_thresh = 1e-09)
```

### Chromplot

Take a closer look at the results by chromosome

```{r}
CHR="chr16"
chromplot(gwas_CD,chr=CHR, variants = snps_CD, annotation_thresh = 5e-09)
```

### Regionplot

Currently the regionplot uses the `gorr` library to fetch genes from Genuity Science' API to get the genes, so platform connection is needed, see Article `Regionplot Example` for an example.

### Manhattan Multi

Display the output from more than one GWAS on the same manhattan plot

```{r}
data(gwas_CD_IIBD)
manhattan(list(gwas_CD_IIBD,gwas_CD), legend_labels=  c("Crohns IIBD", "Crohns UKBB"), alpha = c(1, 0.5))
```
