---
title: "Regionplot Example"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{my-vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Important note

Currently the regionplot uses gor to get the genes, so platform connection is needed

### Load packages

```{r setup}
library(topR)
library(gorr)
library(tidyverse)
```

### Connect to a Genuity Science APIs

First we'll need to establish a connection to  Genuity Science' APIs  To do that we'll need to call `platform_connect`. See full documentation on how to authenticate and establish a connection at [https://wuxi-nextcode.github.io/gorr/](https://wuxi-nextcode.github.io/gorr/)

```{r, eval=FALSE}
api_key <- "your_api_key_here"
conn <- platform_connect(api_key, project = "ukbb_hg38")
conn
```

```{r, echo=FALSE, eval=FALSE}
conn <- platform_connect()
conn
```

If everything goes as planned, we'll have a `conn` object ready to be used for fetching genes.

### Regionplot

# Zoom into a 1 MB region centered on the top variant on the chromosome
top_snp=gwas_CD %>% filter(CHROM == CHR) %>% arrange(P) %>% head(n=1)
#set the 1 MB range centered on the top snp
xmin=top_snp$POS-500000
xmax=top_snp$POS+500000
xmin=ifelse(xmin<0, 0,xmin)
#currently the regionplot uses gor to get the genes, so platform connection is needed
#regionplot(gwas_results,chr=CHR,xmin=xmin,xmax=xmax,variants=lead_snps,annotation_thresh=1e-08,vline=top_snp$POS)

#### Load data
```{r}
data(gwas_CD) 
```

Zoom into a 1 MB region centered on the top variant on the chromosome

```{r, eval=FALSE}
top_snp <- gwas_CD %>%
  filter(CHROM == CHR) %>% 
  arrange(P) %>% 
  head(n = 1)
```

set the 1 MB range centered on the top snp

```{r, eval=FALSE}
xmin <- top_snp$POS - 500000
xmax <- top_snp$POS + 500000
xmin <- ifelse(xmin < 0, 0, xmin)
```

### Plot

```{r, eval=FALSE}
regionplot(gwas_results,
           chr = CHR,
           xmin = xmin,
           xmax = xmax,
           variants = lead_snps,
           annotation_thresh = 1e-08,
           vline = top_snp$POS)
```
