<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">

<style type="text/css">
@font-face {
font-family: octicons-link;
src: url(data:font/woff;charset=utf-8;base64,d09GRgABAAAAAAZwABAAAAAACFQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABEU0lHAAAGaAAAAAgAAAAIAAAAAUdTVUIAAAZcAAAACgAAAAoAAQAAT1MvMgAAAyQAAABJAAAAYFYEU3RjbWFwAAADcAAAAEUAAACAAJThvmN2dCAAAATkAAAABAAAAAQAAAAAZnBnbQAAA7gAAACyAAABCUM+8IhnYXNwAAAGTAAAABAAAAAQABoAI2dseWYAAAFsAAABPAAAAZwcEq9taGVhZAAAAsgAAAA0AAAANgh4a91oaGVhAAADCAAAABoAAAAkCA8DRGhtdHgAAAL8AAAADAAAAAwGAACfbG9jYQAAAsAAAAAIAAAACABiATBtYXhwAAACqAAAABgAAAAgAA8ASm5hbWUAAAToAAABQgAAAlXu73sOcG9zdAAABiwAAAAeAAAAME3QpOBwcmVwAAAEbAAAAHYAAAB/aFGpk3jaTY6xa8JAGMW/O62BDi0tJLYQincXEypYIiGJjSgHniQ6umTsUEyLm5BV6NDBP8Tpts6F0v+k/0an2i+itHDw3v2+9+DBKTzsJNnWJNTgHEy4BgG3EMI9DCEDOGEXzDADU5hBKMIgNPZqoD3SilVaXZCER3/I7AtxEJLtzzuZfI+VVkprxTlXShWKb3TBecG11rwoNlmmn1P2WYcJczl32etSpKnziC7lQyWe1smVPy/Lt7Kc+0vWY/gAgIIEqAN9we0pwKXreiMasxvabDQMM4riO+qxM2ogwDGOZTXxwxDiycQIcoYFBLj5K3EIaSctAq2kTYiw+ymhce7vwM9jSqO8JyVd5RH9gyTt2+J/yUmYlIR0s04n6+7Vm1ozezUeLEaUjhaDSuXHwVRgvLJn1tQ7xiuVv/ocTRF42mNgZGBgYGbwZOBiAAFGJBIMAAizAFoAAABiAGIAznjaY2BkYGAA4in8zwXi+W2+MjCzMIDApSwvXzC97Z4Ig8N/BxYGZgcgl52BCSQKAA3jCV8CAABfAAAAAAQAAEB42mNgZGBg4f3vACQZQABIMjKgAmYAKEgBXgAAeNpjYGY6wTiBgZWBg2kmUxoDA4MPhGZMYzBi1AHygVLYQUCaawqDA4PChxhmh/8ODDEsvAwHgMKMIDnGL0x7gJQCAwMAJd4MFwAAAHjaY2BgYGaA4DAGRgYQkAHyGMF8NgYrIM3JIAGVYYDT+AEjAwuDFpBmA9KMDEwMCh9i/v8H8sH0/4dQc1iAmAkALaUKLgAAAHjaTY9LDsIgEIbtgqHUPpDi3gPoBVyRTmTddOmqTXThEXqrob2gQ1FjwpDvfwCBdmdXC5AVKFu3e5MfNFJ29KTQT48Ob9/lqYwOGZxeUelN2U2R6+cArgtCJpauW7UQBqnFkUsjAY/kOU1cP+DAgvxwn1chZDwUbd6CFimGXwzwF6tPbFIcjEl+vvmM/byA48e6tWrKArm4ZJlCbdsrxksL1AwWn/yBSJKpYbq8AXaaTb8AAHja28jAwOC00ZrBeQNDQOWO//sdBBgYGRiYWYAEELEwMTE4uzo5Zzo5b2BxdnFOcALxNjA6b2ByTswC8jYwg0VlNuoCTWAMqNzMzsoK1rEhNqByEyerg5PMJlYuVueETKcd/89uBpnpvIEVomeHLoMsAAe1Id4AAAAAAAB42oWQT07CQBTGv0JBhagk7HQzKxca2sJCE1hDt4QF+9JOS0nbaaYDCQfwCJ7Au3AHj+LO13FMmm6cl7785vven0kBjHCBhfpYuNa5Ph1c0e2Xu3jEvWG7UdPDLZ4N92nOm+EBXuAbHmIMSRMs+4aUEd4Nd3CHD8NdvOLTsA2GL8M9PODbcL+hD7C1xoaHeLJSEao0FEW14ckxC+TU8TxvsY6X0eLPmRhry2WVioLpkrbp84LLQPGI7c6sOiUzpWIWS5GzlSgUzzLBSikOPFTOXqly7rqx0Z1Q5BAIoZBSFihQYQOOBEdkCOgXTOHA07HAGjGWiIjaPZNW13/+lm6S9FT7rLHFJ6fQbkATOG1j2OFMucKJJsxIVfQORl+9Jyda6Sl1dUYhSCm1dyClfoeDve4qMYdLEbfqHf3O/AdDumsjAAB42mNgYoAAZQYjBmyAGYQZmdhL8zLdDEydARfoAqIAAAABAAMABwAKABMAB///AA8AAQAAAAAAAAAAAAAAAAABAAAAAA==) format('woff');
}
body {
-webkit-text-size-adjust: 100%;
text-size-adjust: 100%;
color: #333;
font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
font-size: 16px;
line-height: 1.6;
word-wrap: break-word;
}
a {
background-color: transparent;
}
a:active,
a:hover {
outline: 0;
}
strong {
font-weight: bold;
}
h1 {
font-size: 2em;
margin: 0.67em 0;
}
img {
border: 0;
}
hr {
box-sizing: content-box;
height: 0;
}
pre {
overflow: auto;
}
code,
kbd,
pre {
font-family: monospace, monospace;
font-size: 1em;
}
input {
color: inherit;
font: inherit;
margin: 0;
}
html input[disabled] {
cursor: default;
}
input {
line-height: normal;
}
input[type="checkbox"] {
box-sizing: border-box;
padding: 0;
}
table {
border-collapse: collapse;
border-spacing: 0;
}
td,
th {
padding: 0;
}
* {
box-sizing: border-box;
}
input {
font: 13px / 1.4 Helvetica, arial, nimbussansl, liberationsans, freesans, clean, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
a {
color: #4078c0;
text-decoration: none;
}
a:hover,
a:active {
text-decoration: underline;
}
hr {
height: 0;
margin: 15px 0;
overflow: hidden;
background: transparent;
border: 0;
border-bottom: 1px solid #ddd;
}
hr:before {
display: table;
content: "";
}
hr:after {
display: table;
clear: both;
content: "";
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 15px;
margin-bottom: 15px;
line-height: 1.1;
}
h1 {
font-size: 30px;
}
h2 {
font-size: 21px;
}
h3 {
font-size: 16px;
}
h4 {
font-size: 14px;
}
h5 {
font-size: 12px;
}
h6 {
font-size: 11px;
}
blockquote {
margin: 0;
}
ul,
ol {
padding: 0;
margin-top: 0;
margin-bottom: 0;
}
ol ol,
ul ol {
list-style-type: lower-roman;
}
ul ul ol,
ul ol ol,
ol ul ol,
ol ol ol {
list-style-type: lower-alpha;
}
dd {
margin-left: 0;
}
code {
font-family: Consolas, "Liberation Mono", Menlo, Courier, monospace;
font-size: 12px;
}
pre {
margin-top: 0;
margin-bottom: 0;
font: 12px Consolas, "Liberation Mono", Menlo, Courier, monospace;
}
.select::-ms-expand {
opacity: 0;
}
.octicon {
font: normal normal normal 16px/1 octicons-link;
display: inline-block;
text-decoration: none;
text-rendering: auto;
-webkit-font-smoothing: antialiased;
-moz-osx-font-smoothing: grayscale;
-webkit-user-select: none;
-moz-user-select: none;
-ms-user-select: none;
user-select: none;
}
.octicon-link:before {
content: '\f05c';
}
.markdown-body:before {
display: table;
content: "";
}
.markdown-body:after {
display: table;
clear: both;
content: "";
}
.markdown-body>*:first-child {
margin-top: 0 !important;
}
.markdown-body>*:last-child {
margin-bottom: 0 !important;
}
a:not([href]) {
color: inherit;
text-decoration: none;
}
.anchor {
display: inline-block;
padding-right: 2px;
margin-left: -18px;
}
.anchor:focus {
outline: none;
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 1em;
margin-bottom: 16px;
font-weight: bold;
line-height: 1.4;
}
h1 .octicon-link,
h2 .octicon-link,
h3 .octicon-link,
h4 .octicon-link,
h5 .octicon-link,
h6 .octicon-link {
color: #000;
vertical-align: middle;
visibility: hidden;
}
h1:hover .anchor,
h2:hover .anchor,
h3:hover .anchor,
h4:hover .anchor,
h5:hover .anchor,
h6:hover .anchor {
text-decoration: none;
}
h1:hover .anchor .octicon-link,
h2:hover .anchor .octicon-link,
h3:hover .anchor .octicon-link,
h4:hover .anchor .octicon-link,
h5:hover .anchor .octicon-link,
h6:hover .anchor .octicon-link {
visibility: visible;
}
h1 {
padding-bottom: 0.3em;
font-size: 2.25em;
line-height: 1.2;
border-bottom: 1px solid #eee;
}
h1 .anchor {
line-height: 1;
}
h2 {
padding-bottom: 0.3em;
font-size: 1.75em;
line-height: 1.225;
border-bottom: 1px solid #eee;
}
h2 .anchor {
line-height: 1;
}
h3 {
font-size: 1.5em;
line-height: 1.43;
}
h3 .anchor {
line-height: 1.2;
}
h4 {
font-size: 1.25em;
}
h4 .anchor {
line-height: 1.2;
}
h5 {
font-size: 1em;
}
h5 .anchor {
line-height: 1.1;
}
h6 {
font-size: 1em;
color: #777;
}
h6 .anchor {
line-height: 1.1;
}
p,
blockquote,
ul,
ol,
dl,
table,
pre {
margin-top: 0;
margin-bottom: 16px;
}
hr {
height: 4px;
padding: 0;
margin: 16px 0;
background-color: #e7e7e7;
border: 0 none;
}
ul,
ol {
padding-left: 2em;
}
ul ul,
ul ol,
ol ol,
ol ul {
margin-top: 0;
margin-bottom: 0;
}
li>p {
margin-top: 16px;
}
dl {
padding: 0;
}
dl dt {
padding: 0;
margin-top: 16px;
font-size: 1em;
font-style: italic;
font-weight: bold;
}
dl dd {
padding: 0 16px;
margin-bottom: 16px;
}
blockquote {
padding: 0 15px;
color: #777;
border-left: 4px solid #ddd;
}
blockquote>:first-child {
margin-top: 0;
}
blockquote>:last-child {
margin-bottom: 0;
}
table {
display: block;
width: 100%;
overflow: auto;
word-break: normal;
word-break: keep-all;
}
table th {
font-weight: bold;
}
table th,
table td {
padding: 6px 13px;
border: 1px solid #ddd;
}
table tr {
background-color: #fff;
border-top: 1px solid #ccc;
}
table tr:nth-child(2n) {
background-color: #f8f8f8;
}
img {
max-width: 100%;
box-sizing: content-box;
background-color: #fff;
}
code {
padding: 0;
padding-top: 0.2em;
padding-bottom: 0.2em;
margin: 0;
font-size: 85%;
background-color: rgba(0,0,0,0.04);
border-radius: 3px;
}
code:before,
code:after {
letter-spacing: -0.2em;
content: "\00a0";
}
pre>code {
padding: 0;
margin: 0;
font-size: 100%;
word-break: normal;
white-space: pre;
background: transparent;
border: 0;
}
.highlight {
margin-bottom: 16px;
}
.highlight pre,
pre {
padding: 16px;
overflow: auto;
font-size: 85%;
line-height: 1.45;
background-color: #f7f7f7;
border-radius: 3px;
}
.highlight pre {
margin-bottom: 0;
word-break: normal;
}
pre {
word-wrap: normal;
}
pre code {
display: inline;
max-width: initial;
padding: 0;
margin: 0;
overflow: initial;
line-height: inherit;
word-wrap: normal;
background-color: transparent;
border: 0;
}
pre code:before,
pre code:after {
content: normal;
}
kbd {
display: inline-block;
padding: 3px 5px;
font-size: 11px;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.pl-c {
color: #969896;
}
.pl-c1,
.pl-s .pl-v {
color: #0086b3;
}
.pl-e,
.pl-en {
color: #795da3;
}
.pl-s .pl-s1,
.pl-smi {
color: #333;
}
.pl-ent {
color: #63a35c;
}
.pl-k {
color: #a71d5d;
}
.pl-pds,
.pl-s,
.pl-s .pl-pse .pl-s1,
.pl-sr,
.pl-sr .pl-cce,
.pl-sr .pl-sra,
.pl-sr .pl-sre {
color: #183691;
}
.pl-v {
color: #ed6a43;
}
.pl-id {
color: #b52a1d;
}
.pl-ii {
background-color: #b52a1d;
color: #f8f8f8;
}
.pl-sr .pl-cce {
color: #63a35c;
font-weight: bold;
}
.pl-ml {
color: #693a17;
}
.pl-mh,
.pl-mh .pl-en,
.pl-ms {
color: #1d3e81;
font-weight: bold;
}
.pl-mq {
color: #008080;
}
.pl-mi {
color: #333;
font-style: italic;
}
.pl-mb {
color: #333;
font-weight: bold;
}
.pl-md {
background-color: #ffecec;
color: #bd2c00;
}
.pl-mi1 {
background-color: #eaffea;
color: #55a532;
}
.pl-mdr {
color: #795da3;
font-weight: bold;
}
.pl-mo {
color: #1d3e81;
}
kbd {
display: inline-block;
padding: 3px 5px;
font: 11px Consolas, "Liberation Mono", Menlo, Courier, monospace;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.task-list-item {
list-style-type: none;
}
.task-list-item+.task-list-item {
margin-top: 3px;
}
.task-list-item input {
margin: 0 0.35em 0.25em -1.6em;
vertical-align: middle;
}
:checked+.radio-label {
z-index: 1;
position: relative;
border-color: #4078c0;
}
.sourceLine {
display: inline-block;
}
code .kw { color: #000000; }
code .dt { color: #ed6a43; }
code .dv { color: #009999; }
code .bn { color: #009999; }
code .fl { color: #009999; }
code .ch { color: #009999; }
code .st { color: #183691; }
code .co { color: #969896; }
code .ot { color: #0086b3; }
code .al { color: #a61717; }
code .fu { color: #63a35c; }
code .er { color: #a61717; background-color: #e3d2d2; }
code .wa { color: #000000; }
code .cn { color: #008080; }
code .sc { color: #008080; }
code .vs { color: #183691; }
code .ss { color: #183691; }
code .im { color: #000000; }
code .va {color: #008080; }
code .cf { color: #000000; }
code .op { color: #000000; }
code .bu { color: #000000; }
code .ex { color: #000000; }
code .pp { color: #999999; }
code .at { color: #008080; }
code .do { color: #969896; }
code .an { color: #008080; }
code .cv { color: #008080; }
code .in { color: #008080; }
</style>
<style>
body {
  box-sizing: border-box;
  min-width: 200px;
  max-width: 980px;
  margin: 0 auto;
  padding: 45px;
  padding-top: 0px;
}
</style>

</head>

<body>

<h1 id="topr--an-r-package-for-viewing-and-annotating-genetic-association-data">topr : an R package for viewing and annotating genetic association data</h1>
<p>See full documentation at <a href="https://wuxi-nextcode.github.io/topr/">https://wuxi-nextcode.github.io/topr/</a> ### Installation</p>
<hr>

<p>Install from CRAN:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;topr&quot;</span>)</span></code></pre></div>
<p>Or from github:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;wuxi-nextcode/topr&quot;</span>)</span></code></pre></div>
<p>And then load the package:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(topr)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span></code></pre></div>
<h2 id="example-usage">Example usage</h2>
<hr>

<p>In this example we demonstrate the basic usage of the topr library.</p>
<h3 id="example-input-datasets">Example input datasets</h3>
<p>The topr package comes with three example datasets; <code>CD_UKBB</code> and <code>UC_UKBB</code> which are subsets of association results (SNPs are filtered on P&lt;1e-03) for Crohn´s disease and Ulcerative colitis from the UK biobank, and <code>CD_FINNGEN</code> which is a subset of association results (P&lt;1e-03) of Crohn’s from FinnGen.</p>
<p>The input dataset must include least three columns (<code>CHROM, POS</code> and <code>P</code>), where naming of the columns is flexible (i.e the chr label can be either chr or chrom and is case insensitive).</p>
<p>The chromosome in the <code>CHROM</code> column can be represented with our without the <i>chr</i> suffix, e.g (chr1 or 1)</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(CD_UKBB) <span class="sc">%&gt;%</span> <span class="fu">select</span>(CHROM,POS,P)</span></code></pre></div>
<p>Input data column names and alternatives understood by topr:</p>
<pre>
<i>Required columns and alternative namings:</i>
CHROM:          CHR,chr,chrom,chromosome,CHROMOSOME
POS:            pos,BP,bp,base_pair_location
P:              p,PVAL,pval,PVALUE,pvalue,p_value

<i>Optional columns and alternatives:</i>
ID:             rsid,rsId,RSID,SNP,snp,rsName,rsname,RSNAME
Gene_Symbol:    Gene_symbol,GENENAME,geneName,genename,GENE,gene
Max_Impact:     max_impact,Impact,impact
REF:            ref
ALT:            alt
BETA:           beta,b,B
OR:             or,odds_ratio
</pre>

<p>It is highly recommended to check the number of datapoints in your dataset before you plot, since a very large dataset will take a long time to plot.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(<span class="st">&quot;Number of SNPs in the dataset: [&quot;</span>, <span class="fu">length</span>(CD_UKBB<span class="sc">$</span>POS),<span class="st">&quot;]&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>If your dataset is very large, you reduce it by filtering variants by their p-values</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gwas_smaller <span class="ot">&lt;-</span> CD_UKBB <span class="sc">%&gt;%</span> <span class="fu">filter</span>(P <span class="sc">&lt;</span> <span class="fl">1e-05</span>)</span></code></pre></div>
<h3 id="manhattan-plots">Manhattan plots</h3>
<p>Get an overview of association results for crohn’s disease (CD) on a Manhattan plot</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(CD_UKBB)</span></code></pre></div>
<h3 id="label-the-top-snps-with-the-name-of-their-nearest-gene">Label the top SNPs with the name of their nearest gene</h3>
<p>Use the <code>annotate</code> argument in the manhattan function to label the top SNPs with p-values below the annotate threshold with their nearest gene</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(CD_UKBB, <span class="at">annotate =</span> <span class="fl">5e-09</span>)</span></code></pre></div>
<p>Change the color, size and angle of the labels. Move the labels along the x and y - axes with the ggrepel parameters (<code>nudge_x</code> and <code>nudge_y</code>)</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(CD_UKBB, <span class="at">annotate =</span> <span class="fl">5e-09</span>, <span class="at">angle=</span><span class="dv">25</span>, <span class="at">label_color=</span><span class="st">&quot;gray40&quot;</span>,<span class="at">label_size=</span><span class="fl">3.5</span>, <span class="at">nudge_x =</span> <span class="dv">100000000</span>, <span class="at">nudge_y=</span><span class="dv">2</span>)</span></code></pre></div>
<h2 id="highlight-genes-of-interest-at-the-base-of-the-plot">Highlight genes of interest at the base of the plot</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(CD_UKBB, <span class="at">annotate =</span> <span class="fl">5e-09</span>, <span class="at">highlight_genes =</span> <span class="fu">c</span>(<span class="st">&quot;NOD2&quot;</span>,<span class="st">&quot;IL23R&quot;</span>,<span class="st">&quot;JAK2&quot;</span>), <span class="at">angle=</span><span class="dv">25</span>)</span></code></pre></div>
<h2 id="the-significance-threshold">The significance threshold</h2>
<p>Remove the significance threshold</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(CD_UKBB, <span class="at">annotate =</span> <span class="fl">5e-09</span>, <span class="at">sign_thresh =</span> <span class="cn">NULL</span>)</span></code></pre></div>
<p>Use multiple significance thresholds and set different colors for each</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(CD_UKBB, <span class="at">annotate =</span> <span class="fl">5e-09</span>, <span class="at">sign_thresh =</span> <span class="fu">c</span>(<span class="fl">1e-07</span>,<span class="fl">1e-09</span>,<span class="fl">1e-11</span>), <span class="at">sign_thresh_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&quot;darkorange&quot;</span>))</span></code></pre></div>
<h3 id="view-one-chromsome-only">View one chromsome only</h3>
<p>Take a closer look at the results by chromosome. Here we plot the results on chromosome 1 only.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(CD_UKBB, <span class="at">annotate =</span> <span class="fl">1e-09</span>, <span class="at">chr =</span> <span class="st">&quot;1&quot;</span>)</span></code></pre></div>
<h3 id="regionplot">Regionplot</h3>
<p>Zoom in further on the chromosome plot with the <code>regionplot</code> function.</p>
<p>Zoom in on a gene of interest, e.g IL23R:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">regionplot</span>(CD_UKBB, <span class="at">gene=</span><span class="st">&quot;IL23R&quot;</span>, <span class="at">annotate=</span> <span class="fl">1e-09</span>)</span></code></pre></div>
<p>Zoom in on the top hit on a chromosome</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>CHR <span class="ot">&lt;-</span> <span class="st">&quot;chr16&quot;</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>top_hit <span class="ot">&lt;-</span> <span class="fu">get_top_hit</span>(CD_UKBB,<span class="at">chr=</span>CHR)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">regionplot</span>(CD_UKBB, <span class="at">chr =</span> CHR, <span class="at">xmin=</span>top_hit<span class="sc">$</span>POS<span class="dv">-150000</span> ,<span class="at">xmax=</span> top_hit<span class="sc">$</span>POS<span class="sc">+</span><span class="dv">150000</span>)</span></code></pre></div>
<h2 id="display-multiple-gwas-results-on-the-same-plot">Display multiple GWAS results on the same plot</h2>
<p>One or more GWAS results can be plotted together on the same plot</p>
<h3 id="manhattan-multiple-phenotypesgwases">Manhattan multiple phenotypes/GWASes</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(UC_UKBB,CD_UKBB,CD_FINNGEN),<span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;UC UKBB&quot;</span>,<span class="st">&quot;CD UKBB&quot;</span>,<span class="st">&quot;CD Finngen&quot;</span>),<span class="at">title=</span><span class="st">&quot;IBD&quot;</span>)</span></code></pre></div>
<p>Annotate top variants</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(UC_UKBB,CD_UKBB,CD_FINNGEN),<span class="at">annotate=</span><span class="fl">1e-12</span>,<span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;UC UKBB&quot;</span>,<span class="st">&quot;CD UKBB&quot;</span>,<span class="st">&quot;CD Finngen&quot;</span>),<span class="at">title=</span><span class="st">&quot;IBD&quot;</span>)</span></code></pre></div>
<p>Set different annotation thresholds for each study</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(UC_UKBB,CD_UKBB,CD_FINNGEN),<span class="at">annotate=</span><span class="fu">c</span>(<span class="fl">1e-15</span>,<span class="fl">1e-12</span>,<span class="fl">5e-9</span>),<span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;UC UKBB&quot;</span>,<span class="st">&quot;CD UKBB&quot;</span>,<span class="st">&quot;CD Finngen&quot;</span>),<span class="at">title=</span><span class="st">&quot;IBD&quot;</span>)</span></code></pre></div>
<h3 id="ntop">ntop</h3>
<p>Use the <code>ntop</code> argument to control how many GWASes appear on the top panel of the plot</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(CD_UKBB,CD_FINNGEN,UC_UKBB),<span class="at">ntop=</span><span class="dv">2</span>,<span class="at">annotate=</span><span class="fu">c</span>(<span class="fl">1e-10</span>,<span class="fl">1e-9</span>,<span class="fl">1e-12</span>),<span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;CD UKBB&quot;</span>,<span class="st">&quot;CD Finngen&quot;</span>,<span class="st">&quot;UC UKBB&quot;</span>),<span class="at">title=</span><span class="st">&quot;IBD&quot;</span>, <span class="at">sign_thresh=</span><span class="fu">c</span>(<span class="fl">5e-07</span>,<span class="fl">5e-09</span>), <span class="at">sign_thresh_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;orange&quot;</span>))</span></code></pre></div>
<h3 id="regionplot-multiple-phenotypesgwases">Regionplot multiple phenotypes/GWASes</h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">regionplot</span>(<span class="fu">list</span>(CD_UKBB,CD_FINNGEN),<span class="at">gene=</span><span class="st">&quot;NOD2&quot;</span>, <span class="at">annotate=</span><span class="fu">c</span>(<span class="fl">1e-15</span>, <span class="fl">1e-09</span>), <span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;UKBB&quot;</span>, <span class="st">&quot;Finngen&quot;</span>), <span class="at">title=</span><span class="st">&quot;Crohn&#39;s disease (CD)&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(CD_UKBB,CD_FINNGEN,UC_UKBB),<span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;CD UKBB&quot;</span>,<span class="st">&quot;CD Finngen&quot;</span>,<span class="st">&quot;UC UKBB&quot;</span>),<span class="at">title=</span><span class="st">&quot;IBD&quot;</span>)</span></code></pre></div>
<h4 id="the-ntop-argument">The ntop argument</h4>
<p>Use the ntop argument to set the number of datasets displayed at the top (default value is 3)</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(CD_UKBB,CD_FINNGEN,UC_UKBB),<span class="at">ntop=</span><span class="dv">2</span>,<span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;CD UKBB&quot;</span>,<span class="st">&quot;CD Finngen&quot;</span>,<span class="st">&quot;UC UKBB&quot;</span>),<span class="at">title=</span><span class="st">&quot;IBD&quot;</span>)</span></code></pre></div>
<h2 id="useful-functions">Useful functions</h2>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_top_hit</span>(CD_UKBB, <span class="at">chr=</span><span class="st">&quot;chr16&quot;</span>, <span class="at">eval=</span><span class="cn">FALSE</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>dat1 <span class="ot">&lt;-</span> <span class="fu">get_best_snp_per_MB</span>(CD_UKBB,<span class="at">thresh =</span> <span class="fl">1e-07</span>, <span class="at">region=</span><span class="dv">1000000</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#get overlapping SNPS overlapping in two datasets</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>overlapping_snps <span class="ot">&lt;-</span> dat1 <span class="sc">%&gt;%</span> <span class="fu">get_overlapping_snps_by_pos</span>(CD_FINNGEN)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>overlapping_snps_matched <span class="ot">&lt;-</span> overlapping_snps <span class="sc">%&gt;%</span> <span class="fu">match_alleles</span>()</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>overl_snps_matched_pos_allele_dat1 <span class="ot">&lt;-</span> overlapping_snps_matched <span class="sc">%&gt;%</span> <span class="fu">flip_to_positive_allele_for_dat1</span>()</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>snpset1 <span class="ot">&lt;-</span> overl_snps_matched_pos_allele_dat1 <span class="sc">%&gt;%</span> <span class="fu">annotate_with_nearest_gene</span>(<span class="at">protein_coding_only =</span> T)</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="co">#or do all this in one go, by calling the create_snpset functon</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>snpset1 <span class="ot">&lt;-</span> <span class="fu">create_snpset</span>(CD_FINNGEN, CD_UKBB, <span class="at">thresh =</span> <span class="fl">1e-06</span>)</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>snpset2 <span class="ot">&lt;-</span> <span class="fu">create_snpset</span>(CD_UKBB, CD_FINNGEN, <span class="at">thresh=</span> <span class="fl">1e-06</span>)</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>e1 <span class="ot">&lt;-</span> <span class="fu">effect_plot</span>(snpset1, <span class="at">pheno_x=</span><span class="st">&quot;CD Finngen&quot;</span>, <span class="at">pheno_y=</span><span class="st">&quot;CD UKBB&quot;</span>,<span class="at">color=</span><span class="fu">get_topr_colors</span>()[<span class="dv">1</span>], <span class="at">gene_label_thresh =</span> <span class="dv">1</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>e2 <span class="ot">&lt;-</span> <span class="fu">effect_plot</span>(snpset2, <span class="at">pheno_x=</span><span class="st">&quot;CD UKBB&quot;</span>, <span class="at">pheno_y=</span><span class="st">&quot;CD Finngen&quot;</span>, <span class="at">color=</span><span class="fu">get_topr_colors</span>()[<span class="dv">2</span>], <span class="at">gene_label_thresh=</span><span class="dv">1</span>,<span class="at">annotate_with =</span> <span class="st">&quot;ID&quot;</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(e1,e2)</span></code></pre></div>
<h3 id="plot-apperance-setting-text-sizes">Plot apperance: setting text sizes</h3>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(CD_UKBB,CD_FINNGEN), <span class="at">annotate=</span><span class="fl">1e-09</span>,<span class="at">axis_title_size =</span> <span class="dv">20</span>,<span class="at">axis_text_size =</span> <span class="dv">16</span>,<span class="at">label_size =</span> <span class="dv">5</span>, <span class="at">title_text_size =</span> <span class="dv">16</span>, <span class="at">legend_text_size =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">regionplot</span>(<span class="fu">list</span>(CD_UKBB,CD_FINNGEN), <span class="at">gene=</span><span class="st">&quot;IKZF1&quot;</span>, <span class="at">vline=</span><span class="dv">50274703</span>,<span class="at">title=</span><span class="st">&quot;CD UKBB&quot;</span>, <span class="at">title_text_size =</span> <span class="dv">16</span>,<span class="at">axis_title_size =</span> <span class="dv">20</span>,<span class="at">axis_text_size =</span> <span class="dv">20</span>,<span class="at">legend_text_size =</span> <span class="dv">20</span>)</span></code></pre></div>
<p>Setting alpha, size and shape</p>

</body>
</html>
