<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Create custom plots for viewing genetic association results using ggplot • topR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Create custom plots for viewing genetic association results using ggplot">
<meta property="og:description" content="topR is a collection of plotting functions for visualizing and exploring genetic association results. Association results from multiple phenotypes can be viewed simultaneously, over the entire genome (Manhattan plot) or in the more detailed regional view.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">topR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wuxi-nextcode/topr/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="topr" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#topr" class="anchor"></a>topR</h1></div>
<p>See full documentation at <a href="https://wuxi-nextcode.github.io/topR/" class="uri">https://wuxi-nextcode.github.io/topR/</a></p>
<div id="installing-from-github-using-devtools" class="section level3">
<h3 class="hasAnchor">
<a href="#installing-from-github-using-devtools" class="anchor"></a>Installing from github using <code>devtools</code>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#devtools::install_github("wuxi-nextcode/topR")</span></code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<p>In this example we demonstrate the basic usage of the topR library.</p>
<div id="load-packages" class="section level3">
<h3 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h3>
<p>First load the <code>topr</code> package, the <code>tidyverse</code> package is recommended in general, but not required for this example</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wuxi-nextcode/topr">topR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="co">#&gt; ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──</span>
<span class="co">#&gt; ✔ ggplot2 3.3.3     ✔ purrr   0.3.4</span>
<span class="co">#&gt; ✔ tibble  3.1.2     ✔ dplyr   1.0.6</span>
<span class="co">#&gt; ✔ tidyr   1.1.3     ✔ stringr 1.4.0</span>
<span class="co">#&gt; ✔ readr   1.4.0     ✔ forcats 0.5.0</span>
<span class="co">#&gt; ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──</span>
<span class="co">#&gt; ✖ dplyr::filter() masks stats::filter()</span>
<span class="co">#&gt; ✖ dplyr::lag()    masks stats::lag()</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel">ggrepel</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="loading-and-exploring-prebuilt-datasets" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-and-exploring-prebuilt-datasets" class="anchor"></a>Loading and exploring prebuilt datasets</h3>
<p>Load the <code>gwas_CD</code> dataset, which is a subset of association results (SNPs with P&lt;1e-03) for Crohn´s disease from the UK biobank.</p>
<p>It is highly recommended to theck the number of datapoints in your dataset before you plot, since a very large dataset will take a long time to plot.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gwas_CD</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gwas_CD</span><span class="op">)</span>
<span class="co">#&gt;   CHROM     POS          ID       REF ALT        SE           P       OR</span>
<span class="co">#&gt; 1  chr1 1006415 rs145588482 TGGCAGCTC   T 0.1540620 0.000468758 0.583384</span>
<span class="co">#&gt; 2  chr1 1006415 rs145588482 TGGCAGCTC   T 0.1540620 0.000468758 0.583384</span>
<span class="co">#&gt; 3  chr1 1007256  rs76233940         G   A 0.1540250 0.000401567 0.579783</span>
<span class="co">#&gt; 4  chr1 1007256  rs76233940         G   A 0.1540250 0.000401567 0.579783</span>
<span class="co">#&gt; 5  chr1 1007256  rs76233940         G   A 0.1540250 0.000401567 0.579783</span>
<span class="co">#&gt; 6  chr1 1341559 rs376494450         C   T 0.0732974 0.000151216 1.320130</span>
<span class="co">#&gt;          AF Gene_Symbol Max_Impact         max_consequence</span>
<span class="co">#&gt; 1 0.0129317  AL645608.3     LOWEST downstream_gene_variant</span>
<span class="co">#&gt; 2 0.0129317       ISG15        LOW          intron_variant</span>
<span class="co">#&gt; 3 0.0130091  AL645608.1     LOWEST downstream_gene_variant</span>
<span class="co">#&gt; 4 0.0130091  AL645608.3     LOWEST downstream_gene_variant</span>
<span class="co">#&gt; 5 0.0130091       ISG15        LOW          intron_variant</span>
<span class="co">#&gt; 6 0.0270627        DVL1        LOW          intron_variant</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Number of SNPs in the dataset: ["</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">gwas_CD</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span>,<span class="st">"]"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="co">#&gt; [1] "Number of SNPs in the dataset: [26821]"</span></code></pre></div>
</div>
<div id="manhattan-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#manhattan-plots" class="anchor"></a>Manhattan plots</h3>
<p>Get an overview of association results for crohn’s disease (CD) in a Manhattan plot</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">gwas_CD</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-7-1.png" width="120%"></p>
</div>
<div id="label-the-top-snps-with-the-name-of-the-nearest-gene" class="section level3">
<h3 class="hasAnchor">
<a href="#label-the-top-snps-with-the-name-of-the-nearest-gene" class="anchor"></a>Label the top SNPs with the name of the nearest gene</h3>
<p>Use the <code>get_best_snp_per_MB</code> function to extract the top/lead SNPs from the crohns dataset. Then include them in the manhattan call function with the variants argument.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snps_CD</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_best_snp_per_MB.html">get_best_snp_per_MB</a></span><span class="op">(</span><span class="va">gwas_CD</span>, thresh <span class="op">=</span> <span class="fl">1e-09</span>, region <span class="op">=</span> <span class="fl">10000000</span><span class="op">)</span>
<span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">gwas_CD</span>, variants<span class="op">=</span><span class="va">snps_CD</span>, annotation_thresh <span class="op">=</span> <span class="fl">1e-09</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-8-1.png" width="120%"> ## Label the position of genes of interest</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span><span class="op">=</span><span class="fu"><a href="reference/get_genes_by_Gene_Symbol.html">get_genes_by_Gene_Symbol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NOD2"</span>,<span class="st">"ULK2"</span>,<span class="st">"THADA"</span><span class="op">)</span><span class="op">)</span>
<span class="va">snps_CD</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_best_snp_per_MB.html">get_best_snp_per_MB</a></span><span class="op">(</span><span class="va">gwas_CD</span>, thresh <span class="op">=</span> <span class="fl">1e-09</span>, region <span class="op">=</span> <span class="fl">10000000</span><span class="op">)</span>
<span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">gwas_CD</span>, variants<span class="op">=</span><span class="va">snps_CD</span>, annotation_thresh <span class="op">=</span> <span class="fl">1e-09</span>, highlight_genes<span class="op">=</span><span class="va">genes</span>,highlight_genes_ypos <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-9-1.png" width="120%"></p>
</div>
<div id="chromplot" class="section level3">
<h3 class="hasAnchor">
<a href="#chromplot" class="anchor"></a>Chromplot</h3>
<p>Take a closer look at the results by chromosome. Here we plot the results on chromosome 16 only.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">CHR</span><span class="op">=</span><span class="st">"chr16"</span>
<span class="fu"><a href="reference/chromplot.html">chromplot</a></span><span class="op">(</span><span class="va">gwas_CD</span>,chr<span class="op">=</span><span class="va">CHR</span>, variants <span class="op">=</span> <span class="va">snps_CD</span>, annotation_thresh <span class="op">=</span> <span class="fl">5e-09</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-10-1.png" width="120%"></p>
</div>
<div id="regionplot" class="section level3">
<h3 class="hasAnchor">
<a href="#regionplot" class="anchor"></a>Regionplot</h3>
<p>Zoom in further on the chromosome plot with the <code>regionplot</code> function.</p>
<p>Zoom in on a gene of interest:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="va">gwas_CD</span>, gene<span class="op">=</span><span class="st">"NOD2"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Zoomed to region: chr16:50593587-50834041"</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-11-1.png" width="120%"></p>
<p>Zoom in on the top hit on a chromosome</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">CHR</span><span class="op">=</span><span class="st">"chr1"</span>
<span class="va">top_hit</span><span class="op">=</span><span class="fu"><a href="reference/get_top_hit.html">get_top_hit</a></span><span class="op">(</span><span class="va">gwas_CD</span>,chr<span class="op">=</span><span class="va">CHR</span><span class="op">)</span>
<span class="va">xmin</span><span class="op">=</span><span class="va">top_hit</span><span class="op">$</span><span class="va">POS</span><span class="op">-</span><span class="fl">250000</span>
<span class="va">xmax</span><span class="op">=</span><span class="va">top_hit</span><span class="op">$</span><span class="va">POS</span><span class="op">+</span><span class="fl">250000</span>
<span class="fu"><a href="reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="va">gwas_CD</span>, chr <span class="op">=</span> <span class="va">CHR</span>, xmin<span class="op">=</span><span class="va">xmin</span>,xmax<span class="op">=</span><span class="va">xmax</span>,show_exons <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="co">#&gt; [1] "Zoomed to region: chr1:66966513-67466513"</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-12-1.png" width="120%"></p>
</div>
<div id="display-multiple-datasets-on-the-same-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#display-multiple-datasets-on-the-same-plot" class="anchor"></a>Display multiple datasets on the same plot</h3>
<p>Display the output from more than one GWAS on the same plot</p>
</div>
<div id="chromplot---multiple-phenotypes" class="section level3">
<h3 class="hasAnchor">
<a href="#chromplot---multiple-phenotypes" class="anchor"></a>Chromplot - multiple phenotypes</h3>
<p>Take a closer look at the results by chromosome. Here we plot the results on chromosome 16 only.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">CHR</span><span class="op">=</span><span class="st">"chr16"</span>
<span class="va">snps_CD_IIBD</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_best_snp_per_MB.html">get_best_snp_per_MB</a></span><span class="op">(</span><span class="va">gwas_CD_IIBD</span>, thresh <span class="op">=</span> <span class="fl">1e-09</span>, region <span class="op">=</span> <span class="fl">10000000</span><span class="op">)</span>
<span class="fu"><a href="reference/chromplot.html">chromplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">gwas_CD_IIBD</span>,<span class="va">gwas_CD</span><span class="op">)</span>,chr<span class="op">=</span><span class="va">CHR</span>, variants <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">snps_CD_IIBD</span>,<span class="va">snps_CD</span><span class="op">)</span>,
          annotation_thresh <span class="op">=</span> <span class="fl">5e-09</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Crohns IIBD"</span>, <span class="st">"Crohns UKBB"</span><span class="op">)</span>, alpha<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.7</span><span class="op">)</span>,size<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1.2</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-13-1.png" width="120%"></p>
</div>
<div id="manhattan---multiple-phenotypes" class="section level3">
<h3 class="hasAnchor">
<a href="#manhattan---multiple-phenotypes" class="anchor"></a>Manhattan - multiple phenotypes</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gwas_CD_IIBD</span><span class="op">)</span>
<span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">gwas_CD_IIBD</span>,<span class="va">gwas_CD</span><span class="op">)</span>, legend_labels<span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Crohns IIBD"</span>, <span class="st">"Crohns UKBB"</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-14-1.png" width="120%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">finngen_K11_CD_STRICT</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">finngen_CHRONSMALL</span><span class="op">)</span>
<span class="va">genes</span><span class="op">=</span><span class="fu"><a href="reference/get_genes_by_Gene_Symbol.html">get_genes_by_Gene_Symbol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NOD2"</span>,<span class="st">"ULK2"</span>,<span class="st">"THADA"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">gwas_CD_IIBD</span>,<span class="va">gwas_CD</span>,<span class="va">finngen_K11_CD_STRICT</span>,<span class="va">finngen_CHRONSMALL</span><span class="op">)</span>,
          ntop<span class="op">=</span><span class="fl">3</span>,
          legend_labels<span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Crohns IIBD"</span>, <span class="st">"Crohns UKBB"</span>,<span class="st">"Crohns Finngen (K11 STRICT)"</span>, <span class="st">"Crohns finngen (CHRONSMALL)"</span><span class="op">)</span>,
          alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.5</span>,<span class="fl">0.3</span>,<span class="fl">1</span><span class="op">)</span>,
          highlight_genes <span class="op">=</span> <span class="va">genes</span>,
          highlight_genes_ypos <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-15-1.png" width="120%"></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/wuxi-nextcode/topr/">https://​github.com/​wuxi-nextcode/​topr/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/wuxi-nextcode/topr/issues">https://​github.com/​wuxi-nextcode/​topr/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>LGPL (&gt;= 3)</small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Genuity Science <br><small class="roles"> Author, maintainer, copyright holder </small>  </li>
<li>Thorhildur Juliusdottir <br><small class="roles"> Author </small>  </li>
<li>Andri Stefansson <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Genuity Science, Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
