<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Create a plot comparing variant effects in two datasets — effectplot • topr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Create a plot comparing variant effects in two datasets — effectplot"><meta property="og:description" content="effectplot()"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">topr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/manhattan.html">Intro into manhattan</a>
    </li>
    <li>
      <a href="../articles/regionplot.html">Intro into regionplot</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/GenuityScience/topr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a plot comparing variant effects in two datasets</h1>
    <small class="dont-index">Source: <a href="https://github.com/GenuityScience/topr/blob/HEAD/R/effectplot.R" class="external-link"><code>R/effectplot.R</code></a></small>
    <div class="hidden name"><code>effectplot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>effectplot()</code></p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">effectplot</span><span class="op">(</span>
  <span class="va">df</span>,
  pheno_x <span class="op">=</span> <span class="st">"x_pheno"</span>,
  pheno_y <span class="op">=</span> <span class="st">"y_pheno"</span>,
  annotate_with <span class="op">=</span> <span class="st">"Gene_Symbol"</span>,
  thresh <span class="op">=</span> <span class="fl">1e-08</span>,
  ci_thresh <span class="op">=</span> <span class="fl">1</span>,
  gene_label_thresh <span class="op">=</span> <span class="fl">1e-08</span>,
  color <span class="op">=</span> <span class="fu"><a href="get_topr_colors.html">get_topr_colors</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,
  scale <span class="op">=</span> <span class="fl">1</span>,
  build <span class="op">=</span> <span class="fl">38</span>,
  label_fontface <span class="op">=</span> <span class="st">"italic"</span>,
  label_family <span class="op">=</span> <span class="st">""</span>,
  nudge_y <span class="op">=</span> <span class="fl">0.001</span>,
  nudge_x <span class="op">=</span> <span class="fl">0.001</span>,
  size <span class="op">=</span> <span class="fl">2</span>,
  segment.size <span class="op">=</span> <span class="fl">0.2</span>,
  segment.linetype <span class="op">=</span> <span class="st">"solid"</span>,
  segment.color <span class="op">=</span> <span class="st">"transparent"</span>,
  angle <span class="op">=</span> <span class="fl">0</span>,
  title <span class="op">=</span> <span class="cn">NULL</span>,
  axis_text_size <span class="op">=</span> <span class="fl">10</span>,
  axis_title_size <span class="op">=</span> <span class="fl">12</span>,
  title_text_size <span class="op">=</span> <span class="fl">13</span>,
  subtitle_text_size <span class="op">=</span> <span class="fl">11</span>,
  gene_label_size <span class="op">=</span> <span class="fl">3.2</span>,
  snpset_thresh <span class="op">=</span> <span class="fl">1e-08</span>,
  snpset_region_size <span class="op">=</span> <span class="fl">1e+06</span>,
  max.overlaps <span class="op">=</span> <span class="fl">10</span>,
  annotate <span class="op">=</span> <span class="fl">0</span>,
  label_color <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>df</dt>
<dd><p>The input dataframe (snpset) containing one row per variant and P values (P1 and P2) and effects (E1 and E2) from two datasets/phenotypes OR a list containing two datasets.</p></dd>
<dt>pheno_x</dt>
<dd><p>A string representing the name of the phenotype whose effect is plotted on the x axis</p></dd>
<dt>pheno_y</dt>
<dd><p>A string representing the name of the phenotype whose effect is plotted on the y axis</p></dd>
<dt>annotate_with</dt>
<dd><p>A string, The name of the column that contains the label for the datapoints (default value is Gene_Symbol)</p></dd>
<dt>thresh</dt>
<dd><p>A number. Threshold cutoff, datapoints with P2 below this threshold are shown as filled circles whereas datapoints with P2 above this threshold are shown as open circles</p></dd>
<dt>ci_thresh</dt>
<dd><p>A number.Show the confidence intervals if the P-value is below this threshold</p></dd>
<dt>gene_label_thresh</dt>
<dd><p>Deprecated: A number, label datapoints with P2 below this threshold</p></dd>
<dt>color</dt>
<dd><p>A string, default value is the first of the topr colors</p></dd>
<dt>scale</dt>
<dd><p>A number, to change the size of the title and axes labels and ticks at the same time (default = 1)</p></dd>
<dt>build</dt>
<dd><p>A number representing the genome build. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p></dd>
<dt>label_fontface</dt>
<dd><p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p></dd>
<dt>label_family</dt>
<dd><p>A stirng or a vector of strings. Label font name (default ggrepel argument is "")</p></dd>
<dt>nudge_y</dt>
<dd><p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>
<dt>nudge_x</dt>
<dd><p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>
<dt>size</dt>
<dd><p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p></dd>
<dt>segment.size</dt>
<dd><p>line segment color (ggrepel argument)</p></dd>
<dt>segment.linetype</dt>
<dd><p>line segment solid, dashed, etc.(ggrepel argument)</p></dd>
<dt>segment.color</dt>
<dd><p>line segment thickness (ggrepel argument)</p></dd>
<dt>angle</dt>
<dd><p>A number, the angle of the text label</p></dd>
<dt>title</dt>
<dd><p>A string to set the plot title</p></dd>
<dt>axis_text_size</dt>
<dd><p>A number, size of the x and y axes tick labels (default: 12)</p></dd>
<dt>axis_title_size</dt>
<dd><p>A number, size of the x and y title labels (default: 12)</p></dd>
<dt>title_text_size</dt>
<dd><p>A number, size of the plot title (default: 13)</p></dd>
<dt>subtitle_text_size</dt>
<dd><p>A number setting the text size of the subtitle (default: 11)</p></dd>
<dt>gene_label_size</dt>
<dd><p>A number setting the size of the gene labels shown at the bottom of the plot</p></dd>
<dt>snpset_thresh</dt>
<dd><p>A number representing the threshold used to create the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p></dd>
<dt>snpset_region_size</dt>
<dd><p>A number representing the region size to use when creating the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p></dd>
<dt>max.overlaps</dt>
<dd><p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p></dd>
<dt>annotate</dt>
<dd><p>A number, label datapoints with p-value below below this number (in the second df) by their nearest gene</p></dd>
<dt>label_color</dt>
<dd><p>A string or a vector of strings. To change the color of the gene or variant labels</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>ggplot object</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># \donttest{</span></span>
<span class="r-in"><span class="fu">effectplot</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-plt img"><img src="effectplot-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span class="co"># }</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

