<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Create a dataframe that can be used as input for making effect plots — get_snpset • topr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Create a dataframe that can be used as input for making effect plots — get_snpset"><meta property="og:description" content="get_snpset()"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">topr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/manhattan.html">Intro into manhattan</a>
    </li>
    <li>
      <a href="../articles/regionplot.html">Intro into regionplot</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/GenuityScience/topr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a dataframe that can be used as input for making effect plots</h1>
    <small class="dont-index">Source: <a href="https://github.com/GenuityScience/topr/blob/HEAD/R/effectplot.R" class="external-link"><code>R/effectplot.R</code></a></small>
    <div class="hidden name"><code>get_snpset.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>get_snpset()</code></p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">get_snpset</span><span class="op">(</span>
  <span class="va">df1</span>,
  <span class="va">df2</span>,
  thresh <span class="op">=</span> <span class="fl">1e-08</span>,
  protein_coding_only <span class="op">=</span> <span class="cn">TRUE</span>,
  region_size <span class="op">=</span> <span class="fl">1e+06</span>,
  verbose <span class="op">=</span> <span class="cn">NULL</span>,
  show_full_output <span class="op">=</span> <span class="cn">FALSE</span>,
  build <span class="op">=</span> <span class="fl">38</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>df1</dt>
<dd><p>The dataframe to extract the top snps from (with p-value below thresh)</p></dd>
<dt>df2</dt>
<dd><p>The dataframe in which to search for overlapping SNPs from dataframe1</p></dd>
<dt>thresh</dt>
<dd><p>A number. P-value threshold, only extract variants with p-values below this threshold (5e-09 by default)</p></dd>
<dt>protein_coding_only</dt>
<dd><p>Logical, set this variable to TRUE to only use protein_coding genes for annotation</p></dd>
<dt>region_size</dt>
<dd><p>An integer (default = 1000000) (or a string represented as 100kb or 1MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p></dd>
<dt>verbose</dt>
<dd><p>Logical, (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p></dd>
<dt>show_full_output</dt>
<dd><p>A logical scalar (default:FALSE). Assign to TRUE to show the full output from this function</p></dd>
<dt>build</dt>
<dd><p>A string, genome build, choose between builds 37 (GRCh37) and 38 (GRCh38) (default is 38)</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Dataframe of overlapping snps (snpset)</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># \donttest{</span></span>
<span class="r-in"><span class="va">CD_UKBB_index_snps</span> <span class="op">&lt;-</span><span class="fu"><a href="get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">get_snpset</span><span class="op">(</span><span class="va">CD_UKBB_index_snps</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $matched</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   CHROM      POS REF1 ALT1          P1        E1 REF2 ALT2          P2       E2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     5 40439961    C    T 7.42668e-11 0.1818381    C    T 2.80350e-13 0.187984</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    16 50485831    A    G 8.18444e-16 0.5591071    A    G 8.06046e-08 0.426680</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3    16 50729867    G   GC 7.36933e-24 0.7618894    G   GC 1.49882e-09 0.530875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4     1 67216513    A    G 8.03684e-20 0.2517527    A    G 1.04321e-08 0.146556</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           ID Gene_Symbol        biotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  rs7713270       TTC33 protein_coding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 rs76176364        NKD1 protein_coding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  rs2066847        NOD2 protein_coding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 rs11576518    C1orf141 protein_coding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $snp_not_found_in_df2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   CHROM       POS REF1 ALT1          P1         E1      ID_tmp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     2 233237298    A    C 1.70216e-09  0.1628839 2_233237298</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2     6  31660620    T    A 1.47082e-24  0.8893100  6_31660620</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3     6  32708532    A    C 1.27821e-15  0.5884587  6_32708532</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4     7  50274703    T    G 8.52335e-14  0.2077781  7_50274703</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5     9   4984530    G    C 5.03745e-11 -0.1833112   9_4984530</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $snp_found_different_alleles_in_df2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] CHROM  POS    REF1   ALT1   P1     E1     REF2   ALT2   P2     E2    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] ID     ID_tmp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;0 rows&gt; (or 0-length row.names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="co"># }</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

