<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Region plot — regionplot • topR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Region plot — regionplot" />
<meta property="og:description" content="


regionplot() displays the association results for a smaler region within one chromosome
Required parameter is at least one dataset (dataframe) containing the association data (with columns CHROM,POS,P in upper or lowercase), xmin and xmax
All other input parameters are optional" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">topR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wuxi-nextcode/topr/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Region plot</h1>
    <small class="dont-index">Source: <a href='https://github.com/wuxi-nextcode/topr/blob/master/R/regionplot.R'><code>R/regionplot.R</code></a></small>
    <div class="hidden name"><code>regionplot.Rd</code></div>
    </div>

    <div class="ref-description">
    


<p><code>regionplot()</code> displays the association results for a smaler region within one chromosome
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase), xmin and xmax</p>
<p>All other input parameters are optional</p>
    </div>

    <pre class="usage"><span class='fu'>regionplot</span><span class='op'>(</span>
  <span class='va'>dat</span>,
  annotation_thresh <span class='op'>=</span> <span class='cn'>NULL</span>,
  title <span class='op'>=</span> <span class='st'>""</span>,
  label_all <span class='op'>=</span> <span class='fl'>0</span>,
  xmin <span class='op'>=</span> <span class='fl'>0</span>,
  size <span class='op'>=</span> <span class='fl'>2</span>,
  shape <span class='op'>=</span> <span class='fl'>19</span>,
  alpha <span class='op'>=</span> <span class='fl'>1</span>,
  label_size <span class='op'>=</span> <span class='fl'>4</span>,
  color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"darkblue"</span>, <span class='st'>"#E69F00"</span>, <span class='st'>"#00AFBB"</span>, <span class='st'>"#999999"</span>, <span class='st'>"#FC4E07"</span>, <span class='st'>"darkorange1"</span><span class='op'>)</span>,
  axis_text_size <span class='op'>=</span> <span class='fl'>11</span>,
  axis_title_size <span class='op'>=</span> <span class='fl'>12</span>,
  title_text_size <span class='op'>=</span> <span class='fl'>13</span>,
  legend_title_size <span class='op'>=</span> <span class='fl'>12</span>,
  legend_text_size <span class='op'>=</span> <span class='fl'>12</span>,
  show_genes <span class='op'>=</span> <span class='cn'>FALSE</span>,
  show_overview <span class='op'>=</span> <span class='cn'>TRUE</span>,
  show_exons <span class='op'>=</span> <span class='cn'>FALSE</span>,
  sign_thresh <span class='op'>=</span> <span class='fl'>5.1e-09</span>,
  sign_thresh_color <span class='op'>=</span> <span class='st'>"red"</span>,
  genes <span class='op'>=</span> <span class='cn'>NULL</span>,
  variants <span class='op'>=</span> <span class='cn'>NULL</span>,
  variant_ids <span class='op'>=</span> <span class='cn'>NULL</span>,
  variant_ids_color <span class='op'>=</span> <span class='st'>"red"</span>,
  xmax <span class='op'>=</span> <span class='cn'>NULL</span>,
  ymin <span class='op'>=</span> <span class='cn'>NULL</span>,
  ymax <span class='op'>=</span> <span class='cn'>NULL</span>,
  protein_coding_only <span class='op'>=</span> <span class='cn'>FALSE</span>,
  chr <span class='op'>=</span> <span class='cn'>NULL</span>,
  vline <span class='op'>=</span> <span class='cn'>NULL</span>,
  legend_name <span class='op'>=</span> <span class='st'>"Data:"</span>,
  legend_position <span class='op'>=</span> <span class='st'>"right"</span>,
  legend_labels <span class='op'>=</span> <span class='cn'>NULL</span>,
  gene <span class='op'>=</span> <span class='cn'>NULL</span>,
  highlight_genes <span class='op'>=</span> <span class='cn'>NULL</span>,
  highlight_genes_color <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>annotation_thresh</th>
      <td><p>Display annotation for variants with p-values below this threshold</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p>Plot title (optional)</p></td>
    </tr>
    <tr>
      <th>xmin</th>
      <td><p>Parameters setting the chromosomal range to display on the x-axis</p></td>
    </tr>
    <tr>
      <th>size</th>
      <td><p>Optional parameter setting the size of the plot points (default: <code>size=1.2</code>)</p></td>
    </tr>
    <tr>
      <th>shape</th>
      <td><p>A number of vector of numers setting the shape of the plotted points</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>A number or vector of numbers setting the transparancy of the plotted points</p></td>
    </tr>
    <tr>
      <th>label_size</th>
      <td><p>Optional parameter to set the size of the plot labels (default: <code>label_size=3.5</code>)</p></td>
    </tr>
    <tr>
      <th>color</th>
      <td><p>Optional parameter setting the color of the plot points (default: <code>color="darkblue"</code>)</p></td>
    </tr>
    <tr>
      <th>axis_text_size</th>
      <td><p>Text size of the x and y axes tick labels (default: 12)</p></td>
    </tr>
    <tr>
      <th>axis_title_size</th>
      <td><p>Text size of the x and y title labels (default: 12)</p></td>
    </tr>
    <tr>
      <th>title_text_size</th>
      <td><p>Text size of the plot title (default: 13)</p></td>
    </tr>
    <tr>
      <th>legend_title_size</th>
      <td><p>Text size of the legend title</p></td>
    </tr>
    <tr>
      <th>legend_text_size</th>
      <td><p>Text size of the legend text</p></td>
    </tr>
    <tr>
      <th>show_genes</th>
      <td><p>Show genes instead of exons (default show_genes=FALSE)</p></td>
    </tr>
    <tr>
      <th>show_overview</th>
      <td><p>Show the overview plot (default show_overview=TRUE)</p></td>
    </tr>
    <tr>
      <th>show_exons</th>
      <td><p>Show exons instead of genees (default show_exons=FALSE)#'</p></td>
    </tr>
    <tr>
      <th>sign_thresh</th>
      <td><p>Optional parameter setting the threshold of the dashed red horizontal line representing the significance threshold (default: <code>sign_thresh=5.1e-9</code>). Multiple thresholds can be provided in a vector, e.g <code>sign_thresh=c(5.1e-9,1.0e-6)</code>). Set this parameter to NULL if you dont want this line to appear at all <code>sign_thresh=NULL</code></p></td>
    </tr>
    <tr>
      <th>sign_thresh_color</th>
      <td><p>set the color of the significance threshold line or lines</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p>List of genes or exons to plot</p></td>
    </tr>
    <tr>
      <th>variants</th>
      <td><p>Optional parameter, a dataset (required columns are <code>CHROM,POS,P</code>) with a list of variants to display and label on the plot.
Variants will be labeled with Gene_Symbol if present in the dataframe, otherwise with ID.  If neither Gene_Symbol nor ID are in the dataset, the variant will be labelled with its position (CHROM and POS)</p></td>
    </tr>
    <tr>
      <th>variant_ids</th>
      <td><p>A vector of variant ids to highlight on the plot, e.g. variant_ids=c("rs1234, rs45898")</p></td>
    </tr>
    <tr>
      <th>variant_ids_color</th>
      <td><p>The color of the variants in variants_id (default color is red)</p></td>
    </tr>
    <tr>
      <th>xmax</th>
      <td><p>Parameters setting the chromosomal range to display on the x-axis</p></td>
    </tr>
    <tr>
      <th>ymin</th>
      <td><p>Optional parameters, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p></td>
    </tr>
    <tr>
      <th>ymax</th>
      <td><p>Optional parameters, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p></td>
    </tr>
    <tr>
      <th>protein_coding_only</th>
      <td><p>Set this parameter to TRUE to only use protein coding genes for annotation</p></td>
    </tr>
    <tr>
      <th>chr</th>
      <td><p>The chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p></td>
    </tr>
    <tr>
      <th>vline</th>
      <td><p>Parameter (optional) to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline=204000066</code>. Multiple values can be provided in a vector, e.g  <code>vline=c(204000066,100500188)</code></p></td>
    </tr>
    <tr>
      <th>legend_name</th>
      <td><p>Change the name of the legend (default: None)</p></td>
    </tr>
    <tr>
      <th>legend_position</th>
      <td><p>Top,bottom,left or right</p></td>
    </tr>
    <tr>
      <th>legend_labels</th>
      <td><p>Legend labels</p></td>
    </tr>
    <tr>
      <th>highlight_genes</th>
      <td><p>A vector of genes or genes to highlight the datapoints for on the plot</p></td>
    </tr>
    <tr>
      <th>highlight_genes_color</th>
      <td><p>Colors for the hihglighted genes (default: green)</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>ggplot object</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='fu'>regionplot</span><span class='op'>(</span><span class='va'>df</span>,xmin<span class='op'>=</span><span class='va'>xmin</span>,xmax<span class='op'>=</span><span class='va'>xmax</span><span class='op'>)</span>
<span class='op'>}</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Genuity Science, Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


