<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Create a regionplot — regionplot • topr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Create a regionplot — regionplot"><meta property="og:description" content="regionplot() displays the association results for a smaller genetic regions within one chromosome.
Required parameter is at least one dataset (dataframe) containing the association data (with columns CHROM,POS,P in upper or lowercase) and either a variant ID, gene name or the genetic region represented as a chromosome together with start and stop positions (either as a single string or as three separate arguments).
All other input parameters are optional"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">topr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/manhattan.html">Intro into manhattan</a>
    </li>
    <li>
      <a href="../articles/regionplot.html">Intro into regionplot</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/wuxi-nextcode/topr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a regionplot</h1>
    <small class="dont-index">Source: <a href="https://github.com/wuxi-nextcode/topr/blob/HEAD/R/regionplot.R" class="external-link"><code>R/regionplot.R</code></a></small>
    <div class="hidden name"><code>regionplot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>regionplot()</code> displays the association results for a smaller genetic regions within one chromosome.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase) and either a variant ID, gene name or the genetic region represented as a chromosome together with start and stop positions (either as a single string or as three separate arguments).</p>
<p>All other input parameters are optional</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">regionplot</span><span class="op">(</span>
  <span class="va">df</span>,
  ntop <span class="op">=</span> <span class="fl">3</span>,
  annotate <span class="op">=</span> <span class="cn">NULL</span>,
  xmin <span class="op">=</span> <span class="fl">0</span>,
  size <span class="op">=</span> <span class="fl">2</span>,
  shape <span class="op">=</span> <span class="fl">19</span>,
  alpha <span class="op">=</span> <span class="fl">1</span>,
  label_size <span class="op">=</span> <span class="fl">4</span>,
  annotate_with <span class="op">=</span> <span class="st">"ID"</span>,
  color <span class="op">=</span> <span class="fu"><a href="get_topr_colors.html">get_topr_colors</a></span><span class="op">(</span><span class="op">)</span>,
  axis_text_size <span class="op">=</span> <span class="fl">11</span>,
  axis_title_size <span class="op">=</span> <span class="fl">12</span>,
  title_text_size <span class="op">=</span> <span class="fl">13</span>,
  show_genes <span class="op">=</span> <span class="cn">FALSE</span>,
  show_overview <span class="op">=</span> <span class="cn">TRUE</span>,
  show_exons <span class="op">=</span> <span class="cn">FALSE</span>,
  max_genes <span class="op">=</span> <span class="fl">200</span>,
  sign_thresh <span class="op">=</span> <span class="fl">5e-09</span>,
  sign_thresh_color <span class="op">=</span> <span class="st">"red"</span>,
  sign_thresh_label_size <span class="op">=</span> <span class="fl">3.5</span>,
  xmax <span class="op">=</span> <span class="cn">NULL</span>,
  ymin <span class="op">=</span> <span class="cn">NULL</span>,
  ymax <span class="op">=</span> <span class="cn">NULL</span>,
  protein_coding_only <span class="op">=</span> <span class="cn">FALSE</span>,
  region_size <span class="op">=</span> <span class="fl">1e+06</span>,
  gene_padding <span class="op">=</span> <span class="fl">1e+05</span>,
  angle <span class="op">=</span> <span class="fl">0</span>,
  legend_title_size <span class="op">=</span> <span class="fl">12</span>,
  legend_text_size <span class="op">=</span> <span class="fl">11</span>,
  nudge_x <span class="op">=</span> <span class="fl">0.01</span>,
  nudge_y <span class="op">=</span> <span class="fl">0.01</span>,
  rsids <span class="op">=</span> <span class="cn">NULL</span>,
  variant <span class="op">=</span> <span class="cn">NULL</span>,
  rsids_color <span class="op">=</span> <span class="cn">NULL</span>,
  legend_name <span class="op">=</span> <span class="st">""</span>,
  legend_position <span class="op">=</span> <span class="st">"right"</span>,
  chr <span class="op">=</span> <span class="cn">NULL</span>,
  vline <span class="op">=</span> <span class="cn">NULL</span>,
  show_gene_names <span class="op">=</span> <span class="cn">NULL</span>,
  legend_labels <span class="op">=</span> <span class="cn">NULL</span>,
  gene <span class="op">=</span> <span class="cn">NULL</span>,
  title <span class="op">=</span> <span class="cn">NULL</span>,
  label_color <span class="op">=</span> <span class="cn">NULL</span>,
  locuszoomplot <span class="op">=</span> <span class="cn">FALSE</span>,
  region <span class="op">=</span> <span class="cn">NULL</span>,
  legend_nrow <span class="op">=</span> <span class="cn">NULL</span>,
  gene_label_size <span class="op">=</span> <span class="cn">NULL</span>,
  scale <span class="op">=</span> <span class="fl">1</span>,
  show_legend <span class="op">=</span> <span class="cn">TRUE</span>,
  sign_thresh_linetype <span class="op">=</span> <span class="st">"dashed"</span>,
  sign_thresh_size <span class="op">=</span> <span class="fl">0.5</span>,
  rsids_with_vline <span class="op">=</span> <span class="cn">NULL</span>,
  annotate_with_vline <span class="op">=</span> <span class="cn">NULL</span>,
  show_gene_legend <span class="op">=</span> <span class="cn">TRUE</span>,
  unit_main <span class="op">=</span> <span class="fl">7</span>,
  unit_gene <span class="op">=</span> <span class="fl">2</span>,
  unit_overview <span class="op">=</span> <span class="fl">1.25</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>df</dt>
<dd><p>Dataframe or a list of dataframes (required columns are <code>CHROM,POS,P</code>), in upper- or lowercase) of association results.</p></dd>
<dt>ntop</dt>
<dd><p>An integer, number of datasets (GWASes) to show on the top plot</p></dd>
<dt>annotate</dt>
<dd><p>A number (p-value). Display annotation for variants with p-values below this threshold</p></dd>
<dt>xmin</dt>
<dd><p>Integer, setting the chromosomal range to display on the x-axis</p></dd>
<dt>size</dt>
<dd><p>An integer setting the size of the plot points (default: <code>size=1.2</code>)</p></dd>
<dt>shape</dt>
<dd><p>A number of vector of numbers setting the shape of the plotted points</p></dd>
<dt>alpha</dt>
<dd><p>A number or vector of numbers setting the transparancy of the plotted points</p></dd>
<dt>label_size</dt>
<dd><p>An number to set the size of the plot labels (default: <code>label_size=3</code>)</p></dd>
<dt>annotate_with</dt>
<dd><p>A string. Annotate the variants with eiher Gene_Symbol or ID (default: "Gene_Symbol")</p></dd>
<dt>color</dt>
<dd><p>A string or a vector of strings, for setting the color of the datapoints on the plot</p></dd>
<dt>axis_text_size</dt>
<dd><p>A number, size of the x and y axes tick labels (default: 12)</p></dd>
<dt>axis_title_size</dt>
<dd><p>A number, size of the x and y title labels (default: 12)</p></dd>
<dt>title_text_size</dt>
<dd><p>A number, size of the plot title (default: 13)</p></dd>
<dt>show_genes</dt>
<dd><p>A logical scalar, show genes instead of exons (default show_genes=FALSE)</p></dd>
<dt>show_overview</dt>
<dd><p>A logical scalar, shows/hides the overview plot (default= TRUE)</p></dd>
<dt>show_exons</dt>
<dd><p>A logical scalar, show exons instead of genees (default show_exons=FALSE)</p></dd>
<dt>max_genes</dt>
<dd><p>An integer, only label the genes if they are fewer than max_genes (default values is 200).</p></dd>
<dt>sign_thresh</dt>
<dd><p>A number or vector of numbers, setting the horizontal signicance threshold (default: <code>sign_thresh=5.1e-9</code>). Set to NULL to hide the significance threshold.</p></dd>
<dt>sign_thresh_color</dt>
<dd><p>A string or vector of strings to set the color/s of the significance threshold/s</p></dd>
<dt>sign_thresh_label_size</dt>
<dd><p>A number setting the text size of the label for the signficance thresholdds (default text size is 3.5)</p></dd>
<dt>xmax</dt>
<dd><p>Integer, setting the chromosomal range to display on the x-axis</p></dd>
<dt>ymin</dt>
<dd><p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p></dd>
<dt>ymax</dt>
<dd><p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p></dd>
<dt>protein_coding_only</dt>
<dd><p>A locical scalar, if TRUE, only protein coding genes are used for annotation</p></dd>
<dt>region_size</dt>
<dd><p>An integer (default = 1000000) indicating the window size for variant labelling. Increase this number for sparser annotation and decrease for denser annotation.</p></dd>
<dt>gene_padding</dt>
<dd><p>An integer representing size of the region around the gene, if the gene argument was used (default = 100000)</p></dd>
<dt>angle</dt>
<dd><p>A number, the angle of the text label</p></dd>
<dt>legend_title_size</dt>
<dd><p>A number, size of the legend title</p></dd>
<dt>legend_text_size</dt>
<dd><p>A number, size of the legend text</p></dd>
<dt>nudge_x</dt>
<dd><p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>
<dt>nudge_y</dt>
<dd><p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>
<dt>rsids</dt>
<dd><p>A string (rsid) or vector of strings to highlight on the plot, e.g. <code>rsids=c("rs1234, rs45898")</code></p></dd>
<dt>variant</dt>
<dd><p>A string representing the variant to zoom in on. Can be either an rsid, or a dataframe (with the columns CHROM,POS,P)</p></dd>
<dt>rsids_color</dt>
<dd><p>A string, the color of the variants in variants_id (default color is red)</p></dd>
<dt>legend_name</dt>
<dd><p>A string, use to change the name of the legend (default: None)</p></dd>
<dt>legend_position</dt>
<dd><p>A string, top,bottom,left or right</p></dd>
<dt>chr</dt>
<dd><p>A string or integer, the chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p></dd>
<dt>vline</dt>
<dd><p>A number or vector of numbers to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline=204000066</code>. Multiple values can be provided in a vector, e.g  <code>vline=c(204000066,100500188)</code></p></dd>
<dt>show_gene_names</dt>
<dd><p>A logical scalar, if set to TRUE, gene names are shown even though they exceed the max_genes count</p></dd>
<dt>legend_labels</dt>
<dd><p>A string or vector of strings representing legend labels for the input dataset/s</p></dd>
<dt>gene</dt>
<dd><p>A string representing the gene to zoom in on (e.g. gene=FTO)</p></dd>
<dt>title</dt>
<dd><p>A string</p></dd>
<dt>label_color</dt>
<dd><p>A string. To change the color of the gene or variant labels</p></dd>
<dt>locuszoomplot</dt>
<dd><p>A logical scalar set to FALSE. Only set to TRUE by calling the locuszoom function</p></dd>
<dt>region</dt>
<dd><p>A string representing a genetic region, e.g. chr1:67038906-67359979</p></dd>
<dt>legend_nrow</dt>
<dd><p>An integer, sets the number of rows allowed for the legend labels</p></dd>
<dt>gene_label_size</dt>
<dd><p>A number setting the size of the gene labels shown at the bottom of the plot</p></dd>
<dt>scale</dt>
<dd><p>A number, to change the size of the title and axes labels and ticks at the same time (default = 1)</p></dd>
<dt>show_legend</dt>
<dd><p>A logical scalar, set to FALSE to hide the legend (default value is TRUE)</p></dd>
<dt>sign_thresh_linetype</dt>
<dd><p>A string, the linetype of the horizontal significance threshold (default = dashed)</p></dd>
<dt>sign_thresh_size</dt>
<dd><p>A number, sets the size of the horizontal significance threshold line (default = 1)</p></dd>
<dt>rsids_with_vline</dt>
<dd><p>A string (rsid) or vector of strings to highlight on the plot with their rsids and vertical lines further highligthing their positions</p></dd>
<dt>annotate_with_vline</dt>
<dd><p>A number (p-value). Display annotation and vertical lines for variants with p-values below this threshold</p></dd>
<dt>show_gene_legend</dt>
<dd><p>A logical scalar, set to FALSE to hide the gene legend (default value is TRUE)</p></dd>
<dt>unit_main</dt>
<dd><p>the height unit of the main plot (default = 7)</p></dd>
<dt>unit_gene</dt>
<dd><p>the height unit of the gene plot (default= 2 )</p></dd>
<dt>unit_overview</dt>
<dd><p>the height unit of the overview plot (default = 1.25)</p></dd>
<dt>verbose</dt>
<dd><p>Logical, set to FALSE to get suppress printed information</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>plots within ggplotGrobs, arranged with egg::gtable_frame</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="fu">regionplot</span><span class="op">(</span><span class="va">CD_UKBB</span>, gene<span class="op">=</span><span class="st">"IL23R"</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

